packages/components/src/navigation/use-create-navigation-tree.js

```diff
@@ -18,6 +23,47 @@ export const useCreateNavigationTree = () => {
 		removeNode: removeMenu,
 	} = useNavigationTreeNodes();
 
+	/**
+	 * Stores direct nested menus of menus
+	 * This makes it easy to traverse menu tree
+	 *
+	 * Key is the menu prop of the menu
+	 * Value is an array of menu keys
+	 */
+	const [ directSubMenus, setDirectSubMenus ] = useState( {} );
```

Comment:
Somehow, I'm not satisfied with this name 😠 Left a quick comment for now.

Reply:
`childMenu`? :)

Reply:
+1 for `childMenu`, especially if what we're traversing is a tree-like structure.

Reply:
The more I think about this data structure, the more I think it's a directed graph instead of a tree.

- A menu is a type of node
- An item is another type of node
- Items have edges that point to _**any**_ menu node via the `navigateToMenu` prop

If so, we may want to rename a few things, including the file name. What do you think @david-szabo97 ?

Reply:
I think "tree" is just a name here and doesn't really mean the structure here. We couldn't come up with a good name so we agreed on this 😄  I'm fine with renaming though, but I don't think it's necessary right now.